### PR TITLE
feat: mass assign attributes

### LINKED ISSUE #16985: [CAL-4477] members view: mass select and add attributes
CleanShot 2024-10-07 at 23 57 22@2x

<sub>[CAL-4477](https://linear.app/calcom/issue/CAL-4477/members-view-mass-select-and-add-attributes)</sub>

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

Implements mass assignment of attributes
Fixes: #16985 
Loom: https://www.loom.com/share/e821740381b644cabf6508f84800a198

TODO: https://linear.app/calcom/issue/CAL-4481/refactor-assignment-of-attributes (Out of scope for this PR) 

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- create an organization
- Create attributes of every type
- Bulk assign to users in your org
-
## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/package.json [hunk 1]
```diff
@@ -105,6 +105,7 @@
     "next-seo": "^6.0.0",
     "next-themes": "^0.2.0",
     "nodemailer": "^6.7.8",
+    "nuqs": "^1.20.0",
```
Comment (sean-brydon) [inline apps/web/package.json:108]: https://nuqs.47ng.com/ 

We've talked about using this a lot before so thought now would be a good time

File: packages/features/users/components/UserTable/BulkActions/MassAssignAttributes.tsx [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+                    </CommandItem>
+                  );
+                })}
+            </CommandGroup>
+          </CommandList>
+        </>
+      );
+    }
+
+    if (showMultiSelectWarning) {
```
Comment (sean-brydon) [inline packages/features/users/components/UserTable/BulkActions/MassAssignAttributes.tsx:182]: Warning to show the behaviour of multi select merging with existing options for that attribute.

File: packages/prisma/migrations/20241008124232_remove_attribute_value_unique/migration.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+-- DropIndex
+DROP INDEX "AttributeOption_attributeId_value_key";
```
Comment (sean-brydon) [inline packages/prisma/migrations/20241008124232_remove_attribute_value_unique/migration.sql:2]: We couldnt unique here - i found out that you couldnt assign the two different member the same value. It has no effect on anything else

**Review by alishaz-polymath — APPROVED** [review]
Let's go. Looking great 🚀 🚀

**Pull request merged**